Wikipedia.js

```diff
@@ -69,14 +71,24 @@ function doWeb(doc, url) {
 		item.encyclopediaTitle = 'Wikipedia, the free encyclopedia';
 	}
 
-	item.url = ZU.xpathText(doc, '//li[@id="t-permalink"]/a/@href');
+	// we don't get a permalink directly on mobile, but it goes into the
+	// "retrieved from" footer
+	let permalink = ZU.xpathText(doc, '//li[@id="t-permalink"]/a/@href')
+		|| attr(doc, '.printfooter a', 'href');
 	var revID;
-	if (item.url) {
-		revID = item.url.match(/[&?]oldid=(\d+)/)[1];
+	if (permalink) {
+		revID = permalink.match(/[&?]oldid=(\d+)/)[1];
 		item.extra = 'Page Version ID: ' + revID;
-		item.url = doc.location.protocol + '//' + doc.location.hostname
-					+ item.url;
-	} else {
+		if (permalink.startsWith('/')) {
+			item.url = doc.location.protocol + '//' + doc.location.hostname
```

Comment:
Wikipedia is always HTTPS, so it won't make a difference, but should just hard-code this to `https://` for cleanliness. (Should always just do that unless a site is known not to support it, which is rare (and inexcusable).)